File: 1399-count-largest-group/1399-count-largest-group.cpp
class Solution {
public:
    int sumOfDigits(int n)
    {
        int res(0);
        while(n > 0)
        {
            res += n % 10;
            n /= 10;
        }
        return res;
    }
    int countLargestGroup(int n) {
        int freq[37] = {};
        int mx = INT_MIN;
        for(int i = 1; i < n+1; i++)
        {
            
            freq[sumOfDigits(i)]++;
            mx = max(freq[sumOfDigits(i)], mx);
        }
        int cnt(0);
        for(int& x : freq)
            if(x == mx) cnt++;
        return cnt;
    }
};

File: 1399-count-largest-group/1399-count-largest-group.py
class Solution:
    def sumOfDigits(self, n):
        res = 0
        while n > 0:
            res += n % 10
            n //= 10
        return res
    
    def countLargestGroup(self, n: int) -> int:
        mp = [0] * 37
        mx = 0
        for i in range(1,n+1):
            mp[self.sumOfDigits(i)] += 1
            mx = max(mx, mp[self.sumOfDigits(i)])
        return mp.count(mx)